### PR TITLE
Enable single block mode

### PR DESCRIPTION
**Pull request by skttl**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes #12927

### Description
This adds Single Block Mode pretty much as described in the discussion.

I've opted for the simple boolean for enabling Single Block Mode, with the addition of a label description stating what is needed for actually enabling Single Block Mode.

When Single Block Mode is enabled, the block gets added automatically when the node is created. There is a property action for replacing the block with one from the clipboard, if there is any blocks in the clipboard.

Single Block Mode outputs a BlockListItem<> instead of the BlockListModel, so you have direct access to the block.

If you convert a "multi" Block list to a Single Block Mode Block List, it will keep the existing blocks, and let you delete all but the first one. So the editor can remove the ones not needed before publishing the page.

To test:
- Try to create a single block mode block list, by creating a block list with just one available block, min-max of 1-1, and Single Block Mode checked.
- Verify that the block is created automatically
- Copy a block, and check that you can replace the current block with the one from the clipboard.

22-10-15---22-11-04---164---CVfaiu71Fw
22-10-15---22-10-35---805---iP0ortg6v2

---
_This item has been added to our backlog [AB#23582](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/23582)_

### CODE REVIEW

File: src/Umbraco.Infrastructure/PropertyEditors/ValueConverters/BlockListPropertyValueConverter.cs [hunk 1]
```diff
@@ -3,24 +3,28 @@
...
     private readonly IProfilingLogger _proflog;
 
-    public BlockListPropertyValueConverter(IProfilingLogger proflog, BlockEditorConverter blockConverter)
+    public BlockListPropertyValueConverter(IProfilingLogger proflog, BlockEditorConverter blockConverter, IContentTypeService contentTypeService)
     {
         _proflog = proflog;
         _blockConverter = blockConverter;
         _blockListEditorDataConverter = new BlockListEditorDataConverter();
+        _contentTypeService = contentTypeService;
     }
```
Comment (Zeegaan) [inline src/Umbraco.Infrastructure/PropertyEditors/ValueConverters/BlockListPropertyValueConverter.cs:28]: This is a breaking change, so we can't do it, I suggest making a new constructor instead, and then calling `: this()` from the old one instead 👍

File: src/Umbraco.Infrastructure/PropertyEditors/ValueConverters/BlockListPropertyValueConverter.cs [hunk 2]
```diff
@@ -29,7 +33,38 @@ public override bool IsConverter(IPublishedPropertyType propertyType)
...
+        var isSingleBlockMode = IsSingleBlockMode(propertyType.DataType);
+        if (isSingleBlockMode)
+        {
+            BlockListConfiguration.BlockConfiguration? block =
+                ConfigurationEditor.ConfigurationAs<BlockListConfiguration>(propertyType.DataType.Configuration)?.Blocks.FirstOrDefault();
+
+            ModelType? contentElementType = block?.ContentElementTypeKey is Guid contentElementTypeKey && _contentTypeService.Get(contentElementTypeKey) is IContentType contentType ? ModelType.For(contentType.Alias) : null;
+            ModelType? settingsElementType = block?.SettingsElementTypeKey is Guid settingsElementTypeKey && _contentTypeService.Get(settingsElementTypeKey) is IContentType settingsType ? ModelType.For(settingsType.Alias) : null;
+
+            if (contentElementType != null)
```
Comment (Zeegaan) [inline src/Umbraco.Infrastructure/PropertyEditors/ValueConverters/BlockListPropertyValueConverter.cs:46]:
```suggestion
            if (contentElementType is not null)
```

File: src/Umbraco.Infrastructure/PropertyEditors/ValueConverters/BlockListPropertyValueConverter.cs [hunk 3]
```diff
@@ -29,7 +33,38 @@ public override bool IsConverter(IPublishedPropertyType propertyType)
...
+            {
+                if (settingsElementType != null)
```
Comment (Zeegaan) [inline src/Umbraco.Infrastructure/PropertyEditors/ValueConverters/BlockListPropertyValueConverter.cs:48]:
```suggestion
                if (settingsElementType is not null)
```

**Review by Zeegaan — CHANGES_REQUESTED** [review]
Found a few nitpicky things, the rest of the code LGTM 👍

File: src/Umbraco.Infrastructure/PropertyEditors/ValueConverters/BlockListPropertyValueConverter.cs (hunk 1 above)
Comment (skttl) [reply src/Umbraco.Infrastructure/PropertyEditors/ValueConverters/BlockListPropertyValueConverter.cs:33]: Like https://github.com/umbraco/Umbraco-CMS/pull/13216/commits/a08bc58abd8601a68aa4d432f6fa1d6197617404 right?

Comment (Zeegaan) [reply src/Umbraco.Infrastructure/PropertyEditors/ValueConverters/BlockListPropertyValueConverter.cs:33]: Yep 🙌

**Comment by skttl** [comment]
@Zeegaan I've fixed the merge conflicts :)

**Comment by Zeegaan** [comment]
Looks good, tests good 🎉

**Pull request merged**